llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -507,6 +507,13 @@ static Decomposition decompose(Value *V,
     }
     Value *Op0;
     Value *Op1;
+
+    if (match(V, m_SExt(m_Value(Op0))))
+      return Op0;
```

Comment:
The equivalent unsigned code only strips the zext, but still falls through to the following code. Should we be doing that here as well? That would allow decomposing something like `sext(add nsw)`.

Reply:
Done.

Reply:
Is there a test for it?

Reply:
Done.